protocols/peer/contracts/Peer.sol

```diff
@@ -50,10 +56,59 @@ contract Peer is IPeer, Ownable {
   ) public {
     swapContract = ISwap(_swapContract);
     if (_peerContractOwner != address(0)) {
-      transferOwnership(_peerContractOwner);
+      whitelist[_peerContractOwner] = true;
```

Comment:
if _peerContractOwner == address(0) you still need to whitelist the owner. I would recommend removing this line, and instead adding the following _after_ the if statement (at the end of the constructor):
`whitelist[owner()] = true;`
This catches both cases.